Data Provider, Calculating Error, OLS Method

zsolt2 opened the pull request:
Create Data provider
Filter incorrect data
Implement Ordinary Least Square algorithm with mean
Calculate Least Absolute Error and Least Square Error

zsolt2 committed:
Data Provider, Calculating Error, OLS Method

SzilvasiPeter committed:
Findings for backend

SzilvasiPeter committed:
Findings for linear regression model algorithm

SzilvasiPeter reviewed (changes requested):
Please corporate the review findings

zsolt2 committed:
Fix findings

zsolt2 committed:
Invalid Data Exception
Create invalid data exception class and throw it from DataProvider.Import(), if the data is not in the correct format.
